pallets/thea-staking/src/lib.rs

```diff
@@ -288,6 +315,76 @@ pub mod pallet {
 			Self::deposit_event(Event::<T>::NetworkRemoved { network });
 			Ok(())
 		}
+
+		/// Allows active relayers report one another for free of any predefined misbehavior
+		/// If number of reporters is >= of given coeficient - slashing for pre-set coeficient
+		/// happens If reporter is part of active set - this call's fee is not apply
+		/// Full weight is payed on error
+		#[pallet::call_index(9)]
+		#[pallet::weight(1_000_000)]
+		pub fn report_offence(
+			origin: OriginFor<T>,
+			network_id: u8,
+			offender: T::AccountId,
+			offence: TheaMisbehavior,
+		) -> DispatchResultWithPostInfo {
+			let reporter = ensure_signed(origin)?;
+			// make sure it's active relayer reporting
+			ensure!(
+				<ActiveRelayers<T>>::get(network_id).iter().any(|(r, _)| r.eq(&reporter)),
+				Error::<T>::NotAnActiveRelayer
+			);
+			// check for re-submit
+			//FIXME: should we charge for sequential report of same offence by same reporter?
+			ensure!(
+				<ReportedOffenders<T>>::get(offender.clone(), offence)
+					.unwrap_or_default()
+					.contains(&reporter),
+				Error::<T>::RepeatedReport
+			);
+			// check if coeficient treshold reached and act
+			let threshold = Self::threshold_slashing_coeficient();
+			let active_relayers = <ActiveRelayers<T>>::get(network_id).len();
+			if let Some(reported) = <ReportedOffenders<T>>::get(offender.clone(), offence) {
+				if reported.len() + 1 + (threshold as usize) >= active_relayers {
+					// slash
+					// <CommitedSlashing<T>> -> store commitment to slash so it can be applyed on
+					let coeficient = match offence {
+						// Severe
+						TheaMisbehavior::UnattendedKeygen | TheaMisbehavior::UnattendedOffline =>
+							Self::severe_slashing_coeficient(),
+						// Moderate
+						_ => Self::moderate_slashing_coeficient(),
+					};
+					// at most 100% will be slashed
+					// era end FIXME: make sure total slash <= offender's stake + reward?
+					<CommitedSlashing<T>>::mutate(&offender, |current_slashing| {
+						let new_percentage = *current_slashing + coeficient;
+						let actual_percentage =
+							if new_percentage >= 100 { 100 } else { new_percentage };
+						*current_slashing = actual_percentage;
+					});
+				} else {
+					// extend storage
+					<ReportedOffenders<T>>::mutate(offender.clone(), offence, |offences| {
+						if let Some(offences) = offences {
+							offences.push(reporter.clone());
+						}
+					});
+				}
+			} else {
+				// register first one
+				<ReportedOffenders<T>>::insert(
+					offender.clone(),
+					offence,
+					[reporter.clone()].to_vec(),
+				);
+			}
+			Self::deposit_event(Event::<T>::OffenceReported { offender, reporter, offence });
+
+			// if not reached and offence is not yet scheduled - store report
```

Comment:
Is it missed comment? can't connect it with next line

Reply:
yes.